pkg/runtime.v2/framework/plugins/registry.go

```diff
@@ -29,7 +30,7 @@ import (
 	"github.com/kubeflow/training-operator/pkg/runtime.v2/framework/plugins/torch"
 )
 
-type Registry map[string]func(ctx context.Context, client client.Client, indexer client.FieldIndexer) (framework.Plugin, error)
+type Registry map[string]func(ctx context.Context, client client.Client, cache cache.Cache, indexer client.FieldIndexer) (framework.Plugin, error)
```

Comment:
Why do we need to pass cache everywhere ?

Reply:
The cache is needed here: https://github.com/kubeflow/training-operator/pull/2330/files#diff-2678f0825d50893b1cc0a6e340510d5530435e83b94253fa220c585cee92f0c8R284-R296 so it's possible to make `PodGroupLimitRangeHandler` and `PodGroupRuntimeClassHandler` strictly typed.

Unfortunately passing the manager cache from the main down to the `ReconcilerBuilder` instances forces to have it passed along all the initialization chain.

Reply:
@astefanutti @tenzen-y Is there a way for us to initialize these watchers without bypassing the cache everywhere ?
E.g. If we add the cache parameter to the runtime registry, we force all of the plugins to implement this argument.